File: ingeneo_technical_test/ingeneo_technical_test/src/main/java/com/example/ingeneo_technical_test/entity/dto/ShipDTO.java
/**
 * 
 */
package com.example.ingeneo_technical_test.entity.dto;

import java.util.Date;

import com.example.ingeneo_technical_test.entity.Client;
import com.example.ingeneo_technical_test.entity.Discount;
import com.example.ingeneo_technical_test.entity.Product;
import com.example.ingeneo_technical_test.entity.Warehouse;
import com.fasterxml.jackson.annotation.JsonInclude;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author Johan Casagua
 *
 */
@Data @AllArgsConstructor @NoArgsConstructor @Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ShipDTO {
	private Long id;
	private Product product;
	private Date registryDate;
	private Date deliveryDate;
	private float shippingPrice;
	private String guideNumber;
	private Client client;
	private Warehouse warehouse;
	private Discount discount;
}

File: frontend/ingeneott/src/app/models/ship.dto.ts
import { ClientDTO } from "./client.dto";
import { ProductDTO } from "./product.dto";
import { WarehouseDTO } from "./warehouse.dto";
import { DiscountDTO } from "./discount.dto";


export class ShipDTO{
    id: number;
    product: ProductDTO;
    registryDate: Date;
    deliveryDate: Date;
    shippingPrice: number;
    guideNumber: string;
    client: ClientDTO;
    warehouse: WarehouseDTO;
    discount: DiscountDTO;
    state: ShipState;
    shippingPriceWithDiscount: number;
}

export enum ShipState{
    REGISTRIED = 'REGISTRIED',
	ON_THE_WAY = 'ON_THE_WAY',
	DELIVERIED = 'DELIVERIED',
	RETURNED = 'RETURNED'
}